File: frontend/src/app/core/models/purchase-order-invoice-report.models.ts
export interface PurchaseOrderInvoiceReportRequest {
  startDate?: Date;
  endDate?: Date;
  searchQuery?: string;
  page: number;
  pageSize: number;
  sortField?: string;
  sortOrder?: number;
}

export interface PurchaseOrderInvoiceReportItem {
  id: number;
  invoiceDate: string | Date;
  invoiceRefNum: string;
  orderRefNo: string;
  invoiceQty: number;
  invoiceAmount: number;
  orderQty: number;
  orderAmount: number;
  pendingQty: number;
  pendingAmount: number;
}

export interface PurchaseOrderInvoiceReportResponse {
  data: PurchaseOrderInvoiceReportItem[];
  totalCount: number;
}

File: backend/src/CWI.Application/DTOs/Reports/PurchaseOrderInvoiceReportModels.cs
using System;
using System.Collections.Generic;

namespace CWI.Application.DTOs.Reports;

/// <summary>
/// Satın Alma Siparişi Fatura Raporu isteği
/// </summary>
public class PurchaseOrderInvoiceReportRequest
{
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public string? SearchQuery { get; set; }
    public int Page { get; set; } = 1;
    public int PageSize { get; set; } = 10;
    public string? SortField { get; set; }
    public int SortOrder { get; set; }
}

/// <summary>
/// Satın Alma Siparişi Fatura Raporu DTO
/// </summary>
public class PurchaseOrderInvoiceReportDto
{
    public long Id { get; set; }
    public DateTime InvoiceDate { get; set; }
    public string InvoiceRefNum { get; set; } = string.Empty;
    public string OrderRefNo { get; set; } = string.Empty;
    public int InvoiceQty { get; set; }
    public decimal InvoiceAmount { get; set; }
    public int OrderQty { get; set; }
    public decimal OrderAmount { get; set; }
    public int PendingQty => OrderQty - InvoiceQty;
    public decimal PendingAmount => OrderAmount - InvoiceAmount;
}

/// <summary>
/// Satın Alma Siparişi Fatura Raporu yanıtı
/// </summary>
public class PurchaseOrderInvoiceReportResponse
{
    public List<PurchaseOrderInvoiceReportDto> Data { get; set; } = new();
    public int TotalCount { get; set; }
}
